godotenv.go

```diff
@@ -209,23 +255,26 @@ func loadFile(filename string, overload bool) error {
 	return nil
 }
 
-func readFile(filename string) (envMap map[string]string, err error) {
+func readFile(filename string, lookupFn func(string)(string, bool)) (envMap map[string]string, err error) {
 	file, err := os.Open(filename)
 	if err != nil {
 		return
 	}
 	defer file.Close()
 
-	return Parse(file)
+	return ParseWithLookup(file, lookupFn)
 }
 
 var exportRegex = regexp.MustCompile(`^\s*(?:export\s+)?(.*?)\s*$`)
 
-func parseLine(line string, envMap map[string]string) (key string, value string, err error) {
+func parseLine(line string, envMap map[string]string, lookupFn func(string)(string, bool)) (key string, value string, err error) {
 	if len(line) == 0 {
 		err = errors.New("zero length string")
 		return
 	}
+	if lookupFn == nil {
+		lookupFn = noLookupFn
+	}
 
 	// ditch the comments (but keep quoted hashes)
 	if strings.Contains(line, "#") {
```

Comment:
This was already there, but looks tricky (thinking of things like `${FOO##something}`) https://pubs.opengroup.org/onlinepubs/009604499/utilities/xcu_chap02.html